packages/snaps-rpc-methods/src/permitted/revokePermissions.ts

```diff
@@ -0,0 +1,92 @@
+import type { JsonRpcEngineEndCallback } from '@metamask/json-rpc-engine';
+import type { PermittedHandlerExport } from '@metamask/permission-controller';
+import type {
+  RevokePermissionsHookParams,
+  RevokePermissionsParams,
+  RevokePermissionsResult,
+} from '@metamask/snaps-sdk';
+import type { Snap, SnapPermissions } from '@metamask/snaps-utils';
+import type { JsonRpcRequest, PendingJsonRpcResponse } from '@metamask/utils';
+import { hasProperty, assert } from '@metamask/utils';
+
+import type { MethodHooksObject } from '../utils';
+
+export type RevokePermissionsHooks = {
+  getSnap: () => Snap | undefined;
+  /**
+   * @returns Null.
+   */
+  revokePermissions: (
+    dynamicPermissions: RevokePermissionsHookParams,
+  ) => Promise<null>;
+};
+
+/**
+ * The `snap_revokePermissions` method implementation.
+ * Revokes dynamic permissions at the runtime.
+ *
+ *
+ * @param request - The JSON-RPC request object.
+ * @param response - The JSON-RPC response object.
+ * @param _next - The `json-rpc-engine` "next" callback. Not used by this
+ * function.
+ * @param end - The `json-rpc-engine` "end" callback.
+ * @param hooks - The RPC method hooks.
+ * @param hooks.getSnap - Get Snap hook.
+ * @param hooks.revokePermissions - Revoke permission hook.
+ */
+async function revokePermissionsImplementation(
+  request: JsonRpcRequest<RevokePermissionsParams>,
+  response: PendingJsonRpcResponse<RevokePermissionsResult>,
+  _next: unknown,
+  end: JsonRpcEngineEndCallback,
+  { getSnap, revokePermissions }: RevokePermissionsHooks,
+): Promise<void> {
+  const snap = getSnap();
+
+  const permissionsToRevoke = request.params as SnapPermissions;
+  const permissionsToRevokeKeys = Object.keys(
+    permissionsToRevoke,
+  ) as RevokePermissionsHookParams;
+
+  const initialPermissions = snap?.manifest.initialPermissions;
+  const dynamicPermissions = snap?.manifest.dynamicPermissions;
+
+  assert(initialPermissions, 'Failed to get Snap initial permissions.');
+  assert(
+    dynamicPermissions,
+    'This Snap have no dynamic permissions specified.',
+  );
+
+  permissionsToRevokeKeys.forEach((permissionKey) => {
+    if (hasProperty(initialPermissions, permissionKey)) {
```

Comment:
Added according to the [SIP-14](https://github.com/MetaMask/SIPs/blob/main/SIPS/sip-14.md) proposal:
```
This RPC method MUST validate that the permissions to be revoked do not contain any permissions specified in initialPermissions. Only dynamicPermissions can be revoked.
```